### PR TITLE
Conversion PearDatabase to yii [IT529]

### PR DESCRIPTION
**Pull request by areksolek**
Conversion PearDatabase to yii [IT529]

### CODE REVIEW

File: vtlib/Vtiger/PackageImport.php [hunk 1]
```diff
@@ -632,8 +632,8 @@ public function importTables($modulenode)
 		if (empty($modulenode->tables) || empty($modulenode->tables->table)) {
 			return;
 		}
-		$adb = \PearDatabase::getInstance();
-		$adb->query('SET FOREIGN_KEY_CHECKS = 0;');
+		$db = \App\Db::getInstance();
+		$db->createCommand('SET FOREIGN_KEY_CHECKS = 0')->execute();
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/PackageImport.php]:
```suggestion
		$db->createCommand()->checkIntegrity(false)->execute();
```

File: vtlib/Vtiger/PackageImport.php [hunk 2]
```diff
@@ -652,11 +652,12 @@ public function importTables($modulenode)
 				} else {
 					\App\Log::trace("SQL: $sql ... ", __METHOD__);
 					Utils::executeQuery($sql);
+					$db->createCommand($sql)->execute();
 					\App\Log::trace('DONE', __METHOD__);
 				}
 			}
 		}
-		$adb->query('SET FOREIGN_KEY_CHECKS = 1;');
+		$db->createCommand('SET FOREIGN_KEY_CHECKS = 1')->execute();
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/PackageImport.php]:
```suggestion
		$db->createCommand()->checkIntegrity(true)->execute();
```

File: vtlib/Vtiger/Utils.php [hunk 3]
```diff
@@ -15,6 +15,8 @@
  */
 class Utils
 {
+	public static $dieOnError = false;
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/Utils.php]: doc

File: vtlib/Vtiger/Utils.php [hunk 4]
```diff
@@ -133,22 +135,17 @@ public static function checkTable($tableName)
...
-		$org_dieOnError = $adb->dieOnError;
-		$adb->dieOnError = false;
-		$sql = 'CREATE TABLE ' . $adb->quote($tablename, false) . ' ' . $criteria;
-		if ($suffixTableMeta !== false) {
-			if ($suffixTableMeta === true) {
-				if ($adb->isMySQL()) {
-					$suffixTableMeta = ' ENGINE=InnoDB DEFAULT CHARSET=utf8';
-				} else {
-				}
+		$orgDieOnError = self::$dieOnError;
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/Utils.php]: remove

File: vtlib/Vtiger/Utils.php [hunk 5]
```diff
@@ -133,22 +135,17 @@ public static function checkTable($tableName)
...
+		$sql = 'CREATE TABLE `' . $tablename . '` ' . $criteria;
+		if (false !== $suffixTableMeta) {
+			if (true === $suffixTableMeta) {
+				$suffixTableMeta = ' ENGINE=InnoDB DEFAULT CHARSET=utf8';
 			}
 			$sql .= $suffixTableMeta;
 		}
-		$adb->query($sql);
-		$adb->dieOnError = $org_dieOnError;
+		\App\Db::getInstance()->createCommand($sql)->execute();
```
Comment (mariuszkrzaczkowski) [inline vtlib/Vtiger/Utils.php]: add try and catch

File: vtlib/Vtiger/PackageImport.php [hunk 6]
```diff
@@ -652,11 +652,12 @@ public function importTables($modulenode)
 				} else {
 					\App\Log::trace("SQL: $sql ... ", __METHOD__);
 					Utils::executeQuery($sql);
```
Comment (rskrzypczak) [inline vtlib/Vtiger/PackageImport.php]:
```suggestion
```

File: vtlib/Vtiger/Utils.php [hunk 7]
```diff
@@ -133,22 +134,18 @@ public static function checkTable($tableName)
 	 */
 	public static function createTable($tablename, $criteria, $suffixTableMeta = false)
```
Comment (rskrzypczak) [inline vtlib/Vtiger/Utils.php]: remove this function

**Pull request merged**